File: 프로그래머스/LV1. 숫자문자열과 영단어.py
# 문자열과 replace

def solution(s):
    dic = {"zero": 0, "one": 1, "two": 2, "three": 3, "four": 4,
           "five": 5, "six": 6, "seven": 7, "eight": 8, "nine": 9}

    while True:
        if s.isdecimal():
            break
        for k in dic.keys():
            if k in s:
                s = s.replace(k, str(dic[k]))
    return int(s)

File: 프로그래머스/JS/LV1. 숫자 문자열과 영단어.js
function solution(s) {
  let number = {
    one: "1",
    two: "2",
    three: "3",
    four: "4",
    five: "5",
    six: "6",
    seven: "7",
    eight: "8",
    nine: "9",
    zero: "0",
  };

  for (let [key, value] of Object.entries(number)) {
    let pattern = new RegExp(`${key}`, "g");
    s = s.replace(pattern, value);
  }
  return parseInt(s);
}
